### PR TITLE
Enhance rust sdk build

### PR DESCRIPTION
**Pull request by vaporos**
Transitions the Rust SDK from using protogen to build.rs.

### CODE REVIEW

File: docker/sawtooth-dev-rust [hunk 1]
```diff
@@ -41,24 +41,28 @@ RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sou
     openssl \
     pkg-config \
     python3-grpcio-tools=1.1.3-1 \
+    unzip \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
 
+RUN curl -OL https://github.com/google/protobuf/releases/download/v3.5.1/protoc-3.5.1-linux-x86_64.zip \
+ && unzip protoc-3.5.1-linux-x86_64.zip -d protoc3
```
Comment (peterschwarz) [inline docker/sawtooth-dev-rust]: Could probably clean up the zip file once you're done with it, for image size reasons.

File: docker/sawtooth-dev-rust [hunk 2]
```diff
@@ -41,24 +41,28 @@ RUN echo "deb http://repo.sawtooth.me/ubuntu/ci xenial universe" >> /etc/apt/sou
...
  && rustup-init -y
 
-EXPOSE 4004/tcp
+#EXPOSE 4004/tcp
 
-ENV PATH=$PATH:/project/sawtooth-core/bin:/root/.cargo/bin
+ENV PATH=$PATH:/protoc3/bin:/project/sawtooth-core/bin:/root/.cargo/bin
 
-RUN cargo install protobuf
+#RUN cargo install protobuf
```
Comment (agunde406) [inline docker/sawtooth-dev-rust]: There are several lines that are commented out but still in the file. These should be removed.

File: docker/sawtooth-dev-rust (hunk 1 above)
Comment (vaporos) [reply docker/sawtooth-dev-rust]: Changed this so it removes the zip file. Also added -sS to curl.

File: docker/sawtooth-dev-rust (hunk 2 above)
Comment (vaporos) [reply docker/sawtooth-dev-rust]: Well, that's embarrassing.  Updated to actually remove those lines.

**Pull request merged**